de.dashup.model/src/main/java/de/dashup/model/service/DashupService.java

```diff
@@ -322,31 +284,105 @@ public void updateSection(User user, String sectionName, int sectionId, int pred
             values.put("section_name", sectionName);
         }
         values.put("predecessor_id", predecessor);
-        values.put("successor_id", successor);
 
         if (!values.isEmpty()) {
             this.database.update(Database.Table.USER_SECTIONS, whereParameters, values);
         }
     }
 
-    public void deleteSection(User user, int section_id) throws SQLException {
+    private void deleteSection(User user, int section_id) throws SQLException {
+        deletePanelsOfSection(section_id);
         Map<String, Object> whereParameters = new HashMap<>();
         whereParameters.put("section_id", section_id);
         whereParameters.put("user_id", user.getId());
         database.delete(Database.Table.USER_SECTIONS, whereParameters);
     }
 
-    public void addSection(User user, String sectionName, int predecessor, int successor) throws SQLException {
-        if (sectionName == null) {
-            sectionName = "New Section";
-        }
+    private void deletePanelsOfSection(int section_id) throws SQLException {
+        Map<String, Object> whereParameters = new HashMap<>();
+        whereParameters.put("section_id", section_id);
+        database.delete(Database.Table.SECTIONS_PANELS, whereParameters);
+
+    }
+
+    private int addSection(User user, String sectionName, int order) throws SQLException, NumberFormatException {
 
         Map<String, Object> values = new HashMap<>();
-        values.put("section_name", sectionName);
+
+        values.put("section_name", Objects.requireNonNullElse(sectionName, "New Section"));
         values.put("user_id", user.getId());
-        values.put("predecessor_id", predecessor);
-        values.put("successor_id", successor);
+        values.put("predecessor_id", order);
 
         this.database.insert(Database.Table.USER_SECTIONS, values);
+
+        JSONObject jsonObject = this.database.get(Database.Table.USER_SECTIONS, values).getJSONObject(0);
+        return jsonObject.getInt("section_id");
+
+    }
+
+    private void addPanel(int section_id, int panel_id, int panel_predecessor, String size) throws SQLException {
+        Map<String, Object> values = new HashMap<>();
+        values.put("section_id", section_id);
+        values.put("panel_id", panel_id);
+        values.put("panel_predecessor", panel_predecessor);
+        values.put("size", size);
+
+        this.database.insert(Database.Table.SECTIONS_PANELS, values);
+
+    }
+
+    public void processLayoutModeChanges(LayoutModeStructure layoutModeStructure, User user) throws SQLException {
+        List<LayoutModeSection> sectionsToDelete = layoutModeStructure.getSectionsToDelete();
```

Comment:
Instead of having a method that handles the changes, we might consider to have several methods that execute each change one by one. This approach is less error-prone and easier to implement.
The undo functionality gets lost, but I think its worth it.